lucene/core/src/java/org/apache/lucene/search/DoubleValuesSource.java

```diff
@@ -456,13 +456,16 @@ public String toString() {
 
     @Override
     public SortField rewrite(IndexSearcher searcher) throws IOException {
-      DoubleValuesSortField rewritten = new DoubleValuesSortField(producer.rewrite(searcher), reverse);
+      DoubleValuesSource rewrittenSource = producer.rewrite(searcher);
+      if (rewrittenSource == producer) {
```

Comment:
This might be better as an object equality check instead of reference equality, I'm not sure.

Reply:
Reference equality is the right check.